colossus/src/main/scala/colossus/protocols/memcache/Memcache.scala

```diff
@@ -291,11 +320,18 @@ object MemcacheReplyParser {
   import colossus.util.Combinators._
   import MemcacheReply._
 
+  val numberPiecesInCasValueReply = 5
+
   def apply() = reply
 
-  def reply = delimitedString(' ', '\r') <~ byte |> { pieces =>
+  def reply = delimitedString(' ', '\r') <~ byte |> { pieces: Seq[String] =>
     pieces.head match {
-      case "VALUE"                   => value(Vector.empty, pieces(1), pieces(2).toInt, pieces(3).toInt)
+      case "VALUE"                   =>
+        if (pieces.size < numberPiecesInCasValueReply) {
+          value(Vector.empty, pieces(1), pieces(2).toInt, pieces(3).toInt)
+        } else {
+          casValue(pieces(1), pieces(2).toInt, pieces(3).toInt, pieces(4).toInt)
```

Comment:
Why in regular value the key is set to Vector.empty but in casValue is set to pieces(1) ?

Reply:
i believe this is because we support a [`MemcacheOps#getAll(...)`](https://github.com/tumblr/colossus/blob/master/colossus/src/main/scala/colossus/protocols/memcache/MemcacheOps.scala#L85) which may return multiple values

Reply:
Uhm i dont think that explain why casValue is not value(Vector.empty, pieces(1), pieces(2).toInt, pieces(3).toInt, pieces(4).toInt) I thought casValue simply adds a piece at the end

Reply:
to support the `MemcacheOps#getAll(...)` functionality, `value(...)` is a recursive method. to avoid intertwining the CAS logic with the unneeded recursion in our case, `casValue(...)` is its own method